internal/xpkg/fetch.go

```diff
@@ -29,11 +30,21 @@ import (
 	v1 "github.com/google/go-containerregistry/pkg/v1"
 	"github.com/google/go-containerregistry/pkg/v1/empty"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sirupsen/logrus"
 	"k8s.io/client-go/kubernetes"
 
 	"github.com/crossplane/crossplane-runtime/pkg/errors"
 )
 
+func init() {
+	// NOTE(hasheddan): we set the logrus package-level logger to discard output
+	// due to the fact that the AWS ECR credential helper uses it to log errors
+	// when parsing registry server URL, which happens any time a package is
+	// pulled from a non-ECR registry.
+	// https://github.com/awslabs/amazon-ecr-credential-helper/issues/308
+	logrus.SetOutput(io.Discard)
+}
```

Comment:
I don't love this, but the error log messages being emitted on every package install is pretty much unacceptable IMO. I have already seen some folks who have tested this out get tripped up by seeing them, so I think this is worth it for the short term.

Reply:
Agreed on both counts. The only alternative I can think of would be disabling this closer to `main()` to avoid the `init()`, but I think I prefer it here closer to the code it affects.